### ISSUE TITLE
A Location-based Feature

### ISSUE THREAD

**Issue by nik-r-cmd** [akshitagupta15june/PetMe#1497]
**Is your feature request related to a problem? Please describe.**
Yes, users currently lack an easy way to find nearby places to adopt pets based on their location.

**Describe the solution you'd like**
To implement a location-based search feature that allows users to enter their city or zip code and displays a list of local pet stores, shelters, and rescue organizations where they can adopt pets.

**Describe alternatives you've considered**
1. Having a dedicated page listing adoption places by state/region (less convenient).
2. Partnering with existing pet adoption platforms and embedding their search (less customizable).

**Comment by NandiniNagothu** [comment]
I would like to implement this location based search feature, would be great if u assign this to me.

**Comment by NandiniNagothu** [comment]
Yes it would be a good feature adding a search based feature based on location and type of pets (for example dogs,cats). would be great if u assign this issue to me under GSSOC'24
